tests/test_nlx_directory.py

```diff
@@ -0,0 +1,55 @@
+import os
+from importlib import reload
+from unittest import mock
+
+from django.test import TestCase
+
+from testapp import settings
+from zgw_consumers.constants import NLXDirectories
+from zgw_consumers.models import NLXConfig
+
+
+class NLXConfigEnvvarTestCase(TestCase):
+    custom_demo = "https://demo.custom.nlx.example/"
+    custom_prod = "https://prod.custom.nlx.example/"
+
+    def _reload_settings(self):
+        reload(settings)
+
+    def test_demo_directory_url_default(self):
+        with mock.patch.dict(os.environ, {}, clear=True):
+            self._reload_settings()
+            config = NLXConfig.get_solo()
+            config.directory = NLXDirectories.demo
+            url = config.directory_url
+            self.assertEqual(
+                url, "https://nlx-directory-ui.commonground.acc.utrecht.nl/"
+            )
+
+    def test_demo_directory_url_overridden_by_envvar(self):
+        with mock.patch.dict(
+            os.environ, {"NLX_DIRECTORY_URL_DEMO": self.custom_demo}, clear=True
+        ):
```

Comment:
If you switch to function based tests, you could use the `monkeypatch` pytest fixture for this:

```python
def test_demo_directory_url_overridden_by_envvar(monkeypatch):
    monkeypatch.setenv("NLX_DIRECTORY_URL_DEMO", "...")
    ...
```